background_isolate_channels/lib/main.dart

```diff
@@ -78,7 +78,7 @@ class _MyHomePageState extends State<MyHomePage> {
     // [sharedPreferencesSet] does not happen before opening the
     // [SimpleDatabase] there has to be a way to refresh
     // [_SimpleDatabaseServer]'s [SharedPreferences] cached values.
-    Future.wait([sharedPreferencesSet, tempDirFuture]).then((values) {
+    [sharedPreferencesSet, tempDirFuture].wait.then((values) {
```

Comment:
Maybe convert this to a record so you can access the values more easily?

https://api.dart.dev/stable/3.0.0/dart-async/FutureRecord2/wait.html

Reply:
That's a good idea - it makes the cast and null assertion go away too.